node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -279,43 +281,71 @@ impl ReceivableDaoReal {
+        // with the high bytes integer being negative
+        let main_sql = "update receivable set balance_high_b = balance_high_b + :balance_high_b, \
                  balance_low_b = balance_low_b + :balance_low_b, last_received_timestamp = :last_received where wallet_address = :wallet";
-                let overflow_update_clause = "update receivable set balance_high_b = :balance_high_b, balance_low_b = :balance_low_b, \
+        let overflow_update_clause = "update receivable set balance_high_b = :balance_high_b, balance_low_b = :balance_low_b, \
                 last_received_timestamp = :last_received where wallet_address = :wallet";
 
-                self.big_int_db_processor.execute(
-                    Either::Right(&xactn),
-                    BigIntSqlConfig::new(
-                        main_sql,
-                        overflow_update_clause,
-                        SQLParamsBuilder::default()
-                            .key(WalletAddress(&transaction.from))
-                            .wei_change(Subtraction("balance", transaction.wei_amount))
-                            .other_params(vec![Param::new(
-                                (":last_received", &to_time_t(timestamp)),
-                                true,
-                            )])
-                            .build(),
-                    ),
-                )?
+        received_payments.iter().for_each(|received_payment| {
+            let last_received = to_time_t(timestamp);
+            let params = SQLParamsBuilder::default()
+                .key(WalletAddress(&received_payment.from))
+                .wei_change(Subtraction("balance", received_payment.wei_amount))
+                .other_params(vec![Param::BothClauses((":last_received", &last_received))])
+                .build();
+
+            let write_result = big_int_db_processor.execute(
+                conn,
+                BigIntSqlConfig::new(main_sql, overflow_update_clause, params),
+            );
+
+            if let Err(e) = write_result {
+                match e {
+                    BigIntDbError::General(err_msg) => panic!("{}", err_msg),
+                    BigIntDbError::RowChangeMismatch { .. } => {
+                        Self::log_or_panic(conn, received_payment, logger)
+                    }
+                }
             }
+        })
+    }
+
+    fn log_or_panic(
+        conn: &dyn ConnectionWrapper,
+        received_payment: &BlockchainTransaction,
+        logger: &Logger,
+    ) {
+        if Self::check_row_presence(conn, &received_payment.from) {
+            panic!(
+                "Update for received payment with {} wei was run without producing a data \
+            change, despite the account for wallet address {} is present",
+                received_payment.wei_amount, received_payment.from
+            )
+        } else {
+            info!(
```

Comment:
There is nothing more we can do here... and the user will know better possibly. By the highest chances, it was him who sent the transaction (in general we want to treat it as a funding tx which means that we will ignore those; and we log it because, at least, we might want to make the user aware that this is happening) :man_shrugging: